### ISSUE TITLE
Feature Request - Support for Transfer-Encoding: chunked

### ISSUE THREAD

**Issue by wilhen01** [a85/POSTMan-Chrome-Extension#3684]
I came across a case recently where I wanted to test sending POST requests using Transfer-Encoding: chunked. It would be cool if POSTMan had the ability to send requests in this manner and control the size of chunks used in the sending process (in bytes maybe). I appreciate this is quite a niche case but it would have saved me writing some code!

**Comment by prashantagarwal** [comment]
@wilhen01 Thanks for suggesting this feature.

**Comment by suyash248** [comment]
Any update on this feature?

**Comment by steven-yates** [comment]
I would really appreciate this feature as well :)

**Comment by Sewmi** [comment]
It would be much appreciated if you could add this feature.

**Comment by shamasis** [comment]
Note: Chunking is supported in underlying runtime and can be bubbled up.

**Comment by shamasis** [comment]
One more point. I might have originally misread the feature request. Postman already supports chunked *responses*. And this one is a a feature-request for sending *request* in chunked format. Correct me if I re-reinterpreted it wrong!!

**Comment by yahya-uddin** [comment]
@shamasis Yes this is a feature request for *SENDING REQUEST* in chuncked format. Not receiving a response in chunked format which I think PostMan already does.

**Comment by lucasdclopes** [comment]
Yes, please! We are currently using another tool for this, but we would prefer to have all of our testing cases in postman.

**Comment by gfinlay** [comment]
@lucasdclopes If postman does not currently support "Transfer-Encoding: chunked" on http POST requests currently, can you please specify what tool you are using that does support this?

**Comment by schusteraj** [comment]
Would really appreciate this feature as well.

**Comment by stefan123t** [comment]
You can select the Code button (right side under Send / Save) to generate code for some runtimes, e.g. curl.
curl will send data with transfer-encoding: chunked if you set the header and leave out the content-length header.
But there is no option for setting an arbitrary chunk-size.
I have not verified what Postman will actually send via Wireshark or whether it uses libcurl in the back.

I therefor do not know if the following is possible, as 
1. curl might do what it wants when it encounters the header, and 
2. Transfer-Encoding is a hop-by-hop header, i.e. it might only be valid until it reaches the web cache or proxy, etc.

I would still like to ask, if it is possible to implement arbitrary chunk-sizes.
I want to test a Web Cache and how it behaves when sending chunked encoding.
My POST request bodies are only 9k in size, hence a configurable 4k chunk size would be ideal for tests.

**Comment by kenander** [comment]
Would really appreciate this feature as well.

**Comment by stefan123t** [comment]
As described above I used the CURL command generated in Postman and added the -H 'Transfer-Encoding: chunked' option. I have verified the transaction by tracing and indeed CURL does switch to chunked encoding.

curl -ivsS --trace --request POST 'http://server:1234/service/endpoint' --header 'SOAPAction: process' --header 'Content-Type: text/xml; charset=utf-8' -H 'Transfer-Encoding: chunked' --data @message 

Here the file message contains my XML SOAP request message.

**Comment by Alsy** [comment]
I too would like this feature

**Comment by epilburus** [comment]
My team would also be grateful if we could have this feature.

**Comment by littlekath25** [comment]
My team would also really appreciate if this feature gets implemented :)

**Comment by pedroans** [comment]
This is a must have feature, but for some reason it's open since 2017. curl is just a workaround, it's not a solution

**Comment by MikePryadko** [comment]
Any news?

**Comment by jchinkle** [comment]
This would be a well received feature.

**Comment by brokemyspoke** [comment]
^+++ please consider; good to know curl supports in the meantime.  thanks @pedroans

**Comment by DinnieJ** [comment]
Is there any update on the feature ?

**Comment by francoisDiagnosticRapid** [comment]
+1

**Comment by imrankhanatsuncity** [comment]
+1

**Comment by joaolce** [comment]
+1

**Comment by isikhi** [comment]
+1 It can be optional

**Comment by murygin** [comment]
+1

**Comment by SoftAdviceJR** [comment]
+1

**Comment by SaeedRz96** [comment]
still, we all are waiting for this feature in 2024 :)

**Comment by billrob** [comment]
> Yes, please! We are currently using another tool for this, but we would prefer to have all of our testing cases in postman.

Hey @lucasdclopes it's been a while.  What is the other tool you are using?

**Comment by JRStolle** [comment]
+1

**Comment by lso5507** [comment]
+1

**Comment by lwillem2234** [comment]
+1

**Comment by kevinrczhang** [comment]
+1 here, would love to see this in Postman

**Comment by smkivuos** [comment]
> Yes, please! We are currently using another tool for this, but we would prefer to have all of our testing cases in postman.

Hi @lucasdclopes - May i know which tool are you using for this testing?

**Comment by stefan123t** [comment]
> You can select the Code button (right side under Send / Save) to generate code for some runtimes, e.g. curl. **curl will send data with transfer-encoding: chunked if you set the header and leave out the content-length header**. *But there is no option for setting an arbitrary chunk-size*. I have not verified what Postman will actually send via Wireshark or whether it uses libcurl in the back.
> 
> I therefor do not know if the following is possible, as
> 1. curl might do what it wants when it encounters the header, and
> 2. Transfer-Encoding is a hop-by-hop header, i.e. it might only be valid until it reaches the web cache or proxy, etc.
> 
> I would still like to ask, if it is possible to **implement arbitrary chunk-sizes**. I want to test a Web Cache and how it behaves when sending chunked encoding. My POST request bodies are only 9k in size, hence a configurable 4k chunk size would be ideal for tests.

I have since had some equally good experience with the following list of tools which are all Open Source too:
* [Bruno](https://www.usebruno.com/) https://github.com/usebruno/bruno
* [Hoppscotch](https://hoppscotch.io/) https://github.com/hoppscotch/hoppscotch
* [Insomnia](https://insomnia.rest/) https://github.com/Kong/insomnia

> As described above I used the CURL command generated in Postman and added the -H 'Transfer-Encoding: chunked' option. I have verified the transaction by tracing and indeed CURL does switch to chunked encoding.
> 
> `curl -ivsS --trace --request POST 'http://server:1234/service/endpoint' --header 'SOAPAction: process' --header 'Content-Type: text/xml; charset=utf-8' -H 'Transfer-Encoding: chunked' --data @message`
> 
> Here the file `message` contains my XML SOAP request message.

You can actually control the size of the buffer used for receiving or sending according to the CURL documentation as @gschier an Insomnia contributor noted [here](https://github.com/Kong/insomnia/issues/1924#issuecomment-587757244):

>> It is indeed ~15kb within Insomnia. I'm not sure why curl would be different but it looks like the option might be configurable which defaults to 16kB: https://curl.haxx.se/libcurl/c/CURLOPT_BUFFERSIZE.html

The max / min may be version dependent according to: https://curl.se/libcurl/c/CURLOPT_UPLOAD_BUFFERSIZE.html
> The maximum size was 512kB until [7.88.0](https://curl.se/ch/7.88.0.html).

You may also emulate Chunk sizes manually by carefully crafting a custom message with arbitrary Chunk sizes.
Wikipedia describes the [Chunked transfer encoding](https://en.wikipedia.org/wiki/Chunked_transfer_encoding) as follows. 

> Each chunk is preceded by its size in bytes. The transmission ends when a zero-length chunk is received. The chunked keyword in the [Transfer-Encoding](https://en.wikipedia.org/wiki/List_of_HTTP_header_fields#transfer-encoding-response-header) header is used to indicate chunked transfer.
> 
> The following example contains three chunks of size 4, 7, and 11 (hexadecimal "B") octets of data.
> `4␍␊Wiki␍␊7␍␊pedia i␍␊B␍␊n ␍␊chunks.␍␊0␍␊␍␊`
>
> ```
> 4
> Wiki
> 
> 7
> pedia i
> 
> B
> n 
> chunks.
> 
> 0
> 
> ```
